D3D12Bundles/D3D12Bundles.cs

```diff
@@ -237,79 +225,97 @@ struct Vertex {
-                SwapEffect = SwapEffect.FlipDiscard,
-                SampleDescription = SampleDescription.Default,
-            };
-
-            // Swap chain needs the queue so that it can force a flush on it.
-            using IDXGISwapChain1 swapChain = factory!.CreateSwapChainForHwnd(mCommandQueue, base.Handle, swapChainDesc);
-            mSwapChain = swapChain.QueryInterface<IDXGISwapChain3>();
-            mFrameIndex = mSwapChain.CurrentBackBufferIndex;
-
-            // Create descriptor heaps.
+#if DEBUG
+            // Configure debug device (if active).
             {
-                // Describe and create a render target view (RTV) descriptor heap.
-                var rtvHeapDesc = new DescriptorHeapDescription {
-                    DescriptorCount = FrameCount,
-                    Type = DescriptorHeapType.RenderTargetView,
-                    Flags = DescriptorHeapFlags.None,
-                };
-                mRtvHeap = mDevice.CreateDescriptorHeap(rtvHeapDesc);
+                using ID3D12InfoQueue? d3dInfoQueue = device.QueryInterfaceOrNull<ID3D12InfoQueue>();
+                if (d3dInfoQueue != null) {
+                    d3dInfoQueue.SetBreakOnSeverity(MessageSeverity.Corruption, true);
+                    d3dInfoQueue.SetBreakOnSeverity(MessageSeverity.Error, true);
+                    var hide = new MessageId[] {
+                        MessageId.MapInvalidNullRange,
+                        MessageId.UnmapInvalidNullRange,
+                        // Workarounds for debug layer issues on hybrid-graphics systems
+                        MessageId.ExecuteCommandListsWrongSwapChainBufferReference,
+                        MessageId.ResourceBarrierMismatchingCommandListType,
+                    };
 
-                mRtvDescriptorSize = mDevice.GetDescriptorHandleIncrementSize(DescriptorHeapType.RenderTargetView);  //Or mRtvHeap.Description.Type
+                    var filter = new InfoQueueFilter {
+                        DenyList = new InfoQueueFilterDescription {
+                            Ids = hide
+                        }
+                    };
+                    d3dInfoQueue.AddStorageFilterEntries(filter);
 
+                    d3dInfoQueue.AddMessage(MessageCategory.Miscellaneous, MessageSeverity.Warning, MessageId.SamplePositionsMismatchRecordTimeAssumedFromClear, $"Hi, from {mName}");
+                    d3dInfoQueue.AddApplicationMessage(MessageSeverity.Warning, $"Hi, from application {mName}");
+                }
+            }
+#endif
 
-                // Describe and create a depth stencil view (DSV) descriptor heap.
-                var dsvHeapDesc = new DescriptorHeapDescription {
-                    DescriptorCount = 1,
-                    Type = DescriptorHeapType.DepthStencilView,
-                    Flags = DescriptorHeapFlags.None,
-                };
-                mDsvHeap = mDevice.CreateDescriptorHeap(dsvHeapDesc);
```

Comment:
All of this code is now ultimately encapsulated in `HwndSwapChainGraphicsPresenter` and its subclasses.